### PR TITLE
Mb new date search index

### PR DESCRIPTION
**Pull request by michaelabuckley**
Replaced the indexes on hfj_spidx_date to satisfy all queries directly from the index.

Also:
- Add new `online()` option to Add and Drop index migrations to avoid locking.
- push down FK definition to allow non-default names.
- turn old index migrations into no-ops.

### CODE REVIEW

File: hapi-fhir-jpa/src/main/java/ca/uhn/fhir/jpa/util/TestUtil.java [hunk 1]
```diff
@@ -282,7 +283,14 @@ private static void scan(AnnotatedElement theAnnotatedElement, Set<String> theNa
 			} else {
 				Validate.notNull(fk);
 				Validate.isTrue(isNotBlank(fk.name()), "Foreign key on " + theAnnotatedElement + " has no name()");
-				Validate.isTrue(fk.name().startsWith("FK_"));
+				List<String> legacySPHibernateFKNames = Arrays.asList(
```
Comment (jamesagnew) [inline hapi-fhir-jpa/src/main/java/ca/uhn/fhir/jpa/util/TestUtil.java:286]: Nice!! Have wanted to do this for ages...

File: hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/DropIndexTask.java [hunk 2]
```diff
@@ -44,63 +45,74 @@
...
 	public DropIndexTask(String theProductVersion, String theSchemaVersion) {
 		super(theProductVersion, theSchemaVersion);
 	}
 
-	static List<String> createDropIndexSql(DriverTypeEnum.ConnectionProperties theConnectionProperties, String theTableName, String theIndexName, DriverTypeEnum theDriverType) throws SQLException {
-		Validate.notBlank(theIndexName, "theIndexName must not be blank");
-		Validate.notBlank(theTableName, "theTableName must not be blank");
+	List<String> generateSql() throws SQLException {
+		Validate.notBlank(myIndexName, "theIndexName must not be blank");
+		Validate.notBlank(getTableName(), "tableName must not be blank");
```
Comment (VK-SMILECDR) [inline hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/DropIndexTask.java:56]: `theIndexName` vs `tableName` consistency?

**Review by VK-SMILECDR — APPROVED** [review]
Overall looks good

File: hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/DropIndexTask.java (hunk 2 above)
Comment (michaelabuckley) [reply hapi-fhir-sql-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/DropIndexTask.java:56]: Ok.

**Pull request merged**